std/algorithm/searching.d

```diff
@@ -3462,6 +3462,59 @@ unittest
 }
 
 /**
+Computes the first occurrence of `range`'s minimum (respectively maximum).
+
+Params:
+    pred = The ordering predicate to use to determine the extremum (minimum or
+           maximum) element.
+    range = The input range to search.
+
+Returns:
+    The position of the first encounter of the minimum (respectively maximum) in `range`.
+ */
+int getMinPos(alias pred = "a < b", Range)(Range range)
+    if (isInputRange!Range && !isInfinite!Range &&
```

Comment:
InputRange is a struct so it is passed by value. The range is consumed inside the function, but when it returns, there won't be any problems.